src/libexpr/flake/lockfile.cc

```diff
@@ -201,13 +202,13 @@ void LockFile::write(const Path & path) const
     writeFile(path, fmt("%s\n", *this));
 }
 
-bool LockFile::isImmutable() const
+std::optional<FlakeRef> LockFile::isUnlocked() const
```

Comment:
The purpose of this function is to determine whether a lock file is unlocked. The return value is merely a hint to provide better error messages.

Reply:
Yes, I get that. But the point is that the signature of the function makes it mostly impossible to know whether having it return `std::nullopt` means `true` or `false`. The only potential hint is that `(bool) std::nullopt == false`, but your comment seems to indicate that it is misleading because `nullopt` should mean true